src/lua/image.c

```diff
@@ -98,6 +99,51 @@ static int drop_cache(lua_State *L)
   return 0;
 }
 
+static int generate_cache(lua_State *L)
+{
+  dt_lua_image_t imgid = 1;
+  luaA_to(L, dt_lua_image_t, &imgid, 1);
+  const gboolean create_dirs = lua_toboolean(L, 2);
+  const int min = luaL_checkinteger(L, 3);
+  const int max = luaL_checkinteger(L, 4);
+  
+  if(create_dirs)
```

Comment:
What happens if create_dirs is false and the directory does not exists?

We still try to create the mipmap cache below.

Reply:
The code I lifted, darktable-generate-cache, runs in a loop over the images you specify or the database.  It does the directory check once up front and then loops through the images creating mipmaps and storing them.  In an effort to speed up the processing, I moved the decision to check/create the directories into the hands of the caller.  The first call you check/create the directories and then the rest of the calls you can turn it off.  It saves several seconds if the amount of mipmaps to create is large (several hundred plus).

If you don't create the directories first, and they don't exist, then the function runs but doesn't do anything. It doesn't crash either.  Looking at the code I think whatever dt_util_test_image_file on line 134 returns is being interpreted as true and therefore the mipmap generation is skipped.  You can test the no directory behavior by changing line 18 of gen_cache.lua to false.

Reply:
Understood, thanks for the explanations.